### PR TITLE
Port :filename.extension/1 to JS

### LINKED ISSUE #505: Port :filename.extension/1 to JS
`-spec extension(Filename) -> file:filename_all() when Filename :: file:name_all().`

### PR DESCRIPTION
**Pull request by mward-sudo**
Closes #505 

## Summary by CodeRabbit

* **New Features**
  * Added file extension extraction supporting binaries, text lists (iolists) and atoms; preserves raw bytes for invalid UTF-8 and returns the same output type as the input.
  * Handles edge cases: no extension, hidden files, multiple dots, trailing slashes, directories, absolute paths, and empty input.

* **Tests**
  * Added comprehensive JS and Elixir test suites covering valid inputs, edge cases, invalid bitstrings, and error conditions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by bartblast** [comment]
@mward-sudo Since this is a draft should I hold off on with reviewing #506, #508, #510 & #502?

**Comment by mward-sudo** [comment]
Yes 👍

**Comment by mward-sudo** [comment]
> Yes 👍

#502 is ready for review. I will take each PR out of draft once they are ready.

**Comment by mward-sudo** [comment]
Waiting for #502 and #534 to land before finalising implementation of `:filename.extension/1`

**Comment by bartblast** [comment]
@mward-sudo Is this ready for review?

**Comment by mward-sudo** [comment]
Not yet. I'll take it out of draft and mark it for review once it's ready.

I'd like some feedback on #510 before I complete this and #508

File: test/elixir/hologram/ex_js_consistency/erlang/filename_test.exs [hunk 1]
```diff
@@ -98,6 +98,98 @@ defmodule Hologram.ExJsConsistency.Erlang.FilenameTest do
     end
   end
 
+  describe "extension/1" do
```
Comment (bartblast) [inline test/elixir/hologram/ex_js_consistency/erlang/filename_test.exs:101]: Could you add edge case tests for:
- "file." → "." (trailing dot is a valid extension)
- ".hidden.txt" → ".txt" (hidden file with extension)
- ".." → "." (double dot)
- "/" → "" (root path)
- "." → "" (current directory)

(remember about adding matching JS tests)

Comment (mward-sudo) [reply test/elixir/hologram/ex_js_consistency/erlang/filename_test.exs:101]: Done

**Comment by bartblast** [comment]
Looks good, thanks @mward-sudo! 👍
